src/Tree.jsx

```diff
@@ -251,12 +251,23 @@ class Tree extends React.Component {
     }
   }
 
+  onDoubleClick(treeNode) {
+    this.props.onDoubleClick({ node: treeNode });
+  }
+
   onSelect(treeNode) {
     const props = this.props;
     const selectedKeys = [...this.state.selectedKeys];
     const eventKey = treeNode.props.eventKey;
     const index = selectedKeys.indexOf(eventKey);
     let selected;
+
+    if (!props.toggleSelect && index > -1) {
```

Comment:
If you don't want to unselect a TreeNode when user click it, just control it with `selectedKeys`, thanks.